astropy/cosmology/io/tests/test_mapping.py

```diff
@@ -129,6 +129,39 @@ def test_tofrom_mapping_move_tofrom_meta(self, cosmo, to_format, from_format):
         assert got == cosmo  # (Doesn't check metadata)
         assert got.meta["mismatching"] == "will error"
 
+    def test_to_mapping_rename_disjoint(self, cosmo, to_format):
+        """Test roundtrip of ``rename`` in ``to/from_mapping()``."""
+        # rename
+        to_rename = {"name": "name"}
+        with pytest.raises(ValueError, match="disjoint"):
+            to_format("mapping", rename=to_rename)
+
+    def test_from_mapping_rename_disjoint(self, cosmo, to_format, from_format):
+        """Test roundtrip of ``rename`` in ``to/from_mapping()``."""
+        # rename
+        m = to_format("mapping")
+
+        with pytest.raises(ValueError, match="disjoint"):
+            from_format(m, format="mapping", rename={"name": "name"})
+
+    def test_tofrom_mapping_rename(self, cosmo, to_format, from_format):
+        """Test roundtrip of ``rename`` in ``to/from_mapping()``."""
+        # rename
+        to_rename = {"name": "cosmo_name"}
+        m = to_format("mapping", rename=to_rename)
+
+        assert "name" not in m.keys()
+        assert "cosmo_name" in m.keys()
+
+        # Wrong names = error
+        with pytest.raises(TypeError, match="there are unused parameters"):
+            from_format(m, format="mapping")
+
+        # Roundtrip. correct names = success
+        from_rename = {v: k for k, v in to_rename.items()}
+        got = from_format(m, format="mapping", rename=from_rename)
+        assert got == cosmo
```

Comment:
I'd prefer if these were in separate tests. The problem with tests that assert too many things is that if an early assert fails then the following asserts are not executed at all.

Reply:
The steps are tested in the prior tests, so except in a parallel environment or one that randomizes the test order, each piece is already tested. This is a roundtrip test but with `assert` along the way so that if it fails it does so close to the actual problem.

Reply:
SGTM.

Reply:
Done!